### PR TITLE
gui: Refactor notice handling and app selection. 

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- gui/information bar: Small refactor of notice.
Switch to select app from open it for go to new app installed.
Add erase app notice for erase notice of app when delete app corresponding.
Refactor both structure for remove double variable.

### CODE REVIEW

File: vita3k/gui/src/information_bar.cpp [hunk 1]
```diff
@@ -37,31 +37,57 @@
...
+            // Find and erase the corresponding entry in notice_info
+            const auto notice_info_it = std::find_if(notice_info.begin(), notice_info.end(), [&](const NoticeInfo &n) {
+                return n.time == notice_list_it->time;
+            });
+            if (notice_info_it != notice_info.end())
+                notice_info.erase(notice_info_it); // Erase the entry from notice_info
+
+            // Erase the entry from notice_info_icon if it exists
+            if (gui.notice_info_icon.contains(notice_list_it->time))
+                gui.notice_info_icon.erase(notice_list_it->time);
```
Comment (Macdu) [inline vita3k/gui/src/information_bar.cpp:80]:
```suggestion
                gui.notice_info_icon.erase(notice_list_it->time);
```

File: vita3k/gui/src/information_bar.cpp [hunk 2]
```diff
@@ -37,31 +37,57 @@
...
+
+    // Check if the notice list is empty
+    if (notice_global.empty()) {
+        LOG_WARN("Notice list is empty.");
+        return;
+    }
+
+    auto notice_list_it = notice_global.begin();
+    while (notice_list_it != notice_global.end()) {
+        if (notice_list_it->id == title_id) {
```
Comment (Macdu) [inline vita3k/gui/src/information_bar.cpp:70]:
```suggestion
        if (notice_list_it->id != title_id) {
              ++notice_list_it;
              continue;
       }
```

File: vita3k/gui/src/information_bar.cpp [hunk 3]
```diff
@@ -185,10 +208,8 @@ void init_notice_info(GuiState &gui, EmuEnvState &emuenv) {
                     if (!set_notice_info(gui, emuenv, *notice_it)) {
                         notice_it = user.second.erase(notice_it);
                         save_notice_list(emuenv);
-                    } else {
-                        notice_info_new[notice_it->time] = notice_list_new[user.first][notice_it->time];
+                    } else
```
Comment (Macdu) [inline vita3k/gui/src/information_bar.cpp:211]:
```suggestion
                    } else {
```
If the if has brackets, then the else should have brackets too

File: vita3k/gui/src/information_bar.cpp [hunk 4]
```diff
@@ -302,11 +321,16 @@ void update_notice_info(GuiState &gui, EmuEnvState &emuenv, const std::string &t
...
     notice_info_count_new = 0;
-    notice_info_new.clear();
     notice_list_count_new["global"] = 0;
     notice_list_count_new[user_id] = 0;
-    notice_list_new["global"].clear();
-    notice_list_new[user_id].clear();
+    for (auto &notice : notice_info)
+        notice.is_new = false;
+    for (auto &notice : notice_list) {
+        if ((notice.first == "global") || (notice.first == user_id)) {
```
Comment (Macdu) [inline vita3k/gui/src/information_bar.cpp:329]:
```suggestion
    for (auto &[user, lists] : notice_list) {
        if (user== "global" || user== user_id) {
```

File: vita3k/gui/src/information_bar.cpp [hunk 5]
```diff
@@ -379,7 +403,7 @@ static void draw_notice_info(GuiState &gui, EmuEnvState &emuenv) {
     }
 
     if (ImGui::IsWindowHovered(ImGuiHoveredFlags_RootWindow) && !ImGui::IsAnyItemHovered() && ImGui::IsMouseClicked(ImGuiMouseButton_Left)) {
-        if (notice_info_state) {
+        if (notice_info_state && (notice_info_count_new > 0)) {
```
Comment (Macdu) [inline vita3k/gui/src/information_bar.cpp:406]:
```suggestion
        if (notice_info_state && notice_info_count_new > 0) {
```

File: vita3k/gui/src/information_bar.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/gui/src/information_bar.cpp:80]: no get crash if element is missing on map ?

File: vita3k/gui/src/information_bar.cpp (hunk 5 above)
Comment (Zangetsu38) [reply vita3k/gui/src/information_bar.cpp:406]: for that is normal, is formating code of vit'a3k
when compare have more one condition, need ading ( ) if conjdinition have two things with compare operator

**Comment by Zangetsu38** [comment]
Done all normally

**Comment by Zangetsu38** [comment]
Is normally ready 🤔

File: vita3k/gui/src/information_bar.cpp (hunk 5 above)
Comment (Macdu) [reply vita3k/gui/src/information_bar.cpp:406]: This isn't part of code formatting. Without these parenthesis, the formatter doesn't complain, no warning is generated and the code is still right as comparisons always have priority over logical operators, but anyway.

**Review by Macdu — APPROVED** [review]

**Pull request merged**